### PR TITLE
Added typings for ApplicationAPI

### LINKED ISSUE #360: Application Types
Create typings for Application API

### PR DESCRIPTION
**Pull request by bpedersen**
Added typings to ApplicationAPI

## Description
Added typings for the ApplicationAPI from what I could find online

## Motivation and Context
Hacktoberfest (https://developer.nexmo.com/hacktoberfest)
This was requested by https://github.com/Vonage/vonage-node-sdk/issues/360

## Testing Details
No tests should be affected

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by bpedersen** [comment]
Thank you for the Hacktoberfest tag, however it looks like they changed the requirements recently.
Looks like we need to use a tag for PRs as Hacktoberfest-Accepted
https://hacktoberfest.digitalocean.com/hacktoberfest-update

**Comment by mheap** [comment]
Good news! The `vonage-node-sdk` repo has the `hacktoberfest` topic, so all PRs submitted against this repo are implicitly eligible to be counted

File: typings/index.d.ts [hunk 1]
```diff
@@ -1,215 +1,307 @@
...
+		instance: string;
+		invalid_parameters?: { name: string; reason: string }[];
+	}
+
+	export interface AppUrl {
+		address: string;
+		http_method: "POST" | "GET";
+	}
+
+	export interface AppCapabilities {
```
Comment (kellyjandrews) [inline typings/index.d.ts:245]: I think this section is missing a few items - looks like you would need voice, rtc, messages, and vbc as high level objects. Check this section of the API docs - https://developer.nexmo.com/api/application.v2#createApplication

File: typings/index.d.ts [hunk 2]
```diff
@@ -1,215 +1,307 @@
...
+
+	/* Vonage */
+	export default class Vonage {
+		constructor(
+			credentials: CredentialsObject,
+			options?: { [key: string]: any }
+		);
+		public readonly verify: Verify;
+		public readonly message: Message;
+	}
```
Comment (kellyjandrews) [inline typings/index.d.ts:306]: You will need to export the interface like Verify and Message here.

**Review by kellyjandrews — CHANGES_REQUESTED** [review]

**Pull request closed without merging**